src/functions.cpp

```diff
@@ -1,9 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"
 #include <iostream>
 #include <limits>
+#include <algorithm>
 
 void print(std::vector<int> const & numbers) {
-    for (int i = 0; i < numbers.size(); ++i) {
+    for (auto i = 0; i < numbers.size(); ++i) {
```

Comment:
i think the better solution is to use directly std::size_t

Reply:
not sure why size_t, auto seams to be safer, and probabli it will deduce size_t, and program will take the same time as size_t

Reply:
Auto will deduce int here, but proper size is size_t, which is some kind of unsigned type.